tests/generation/test_generation_utils.py

```diff
@@ -1976,6 +1977,12 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwa
             [1, 18],
         )
 
+    def test_stop_sequence_stopping_criteria(self):
+        prompt = """Hello I believe in"""
+        generator = pipeline("text-generation", model="hf-internal-testing/tiny-random-bart")
+        output = generator(prompt, stop_sequence=" number")
+        self.assertEqual(output[0]["generated_text"].split()[-1], "number")
+
```

Comment:
```suggestion
    def test_stop_sequence_stopping_criteria(self):
        prompt = """Hello I believe in"""
        generator = pipeline("text-generation", model="hf-internal-testing/tiny-random-bart")
        output = generator(prompt)
        self.assertEqual(output, [{'generated_text': 'Hello I believe in in in number number number number number number number number number'}])
        
        output = generator(prompt, stop_sequence=" number")
        self.assertEqual(output, [{'generated_text': 'Hello I believe in in in number'}])
```

I think this formulation conveys the intent of the test a tiny bit better.
Also since you were only testing the last generated token, if we deactivated the whole option your test would still pass since the model just generates `number` all the time.
Wdyt?

Reply:
this makes sense. I'll change it to that.